clang/lib/AST/CommentParser.cpp

```diff
@@ -149,6 +276,93 @@ class TextTokenRetokenizer {
     addToken();
   }
 
+  /// Extract a type argument
+  bool lexType(Token &Tok) {
+    if (isEnd())
+      return false;
+    Position SavedPos = Pos;
+    consumeWhitespace();
+    SmallString<32> NextToken;
+    SmallString<32> WordText;
+    const char *WordBegin = Pos.BufferPtr;
+    SourceLocation Loc = getSourceLocation();
+    StringRef ConstVal = StringRef("const");
+    StringRef PointerVal = StringRef("*");
```

Comment:
What about pointers to members? Arrays? Function? More complex declarators like `void(*)(int, int)`?

Reply:
For now, this doesn't support those values, and due to the complexity of implementing them and the rarity of their use I think it'd make more sense to add them in a follow-up PR. Let me know if this sounds reasonable